public/install/functions.php

```diff
@@ -17,126 +25,196 @@ function checkPhpVersion()
+ * @param string $envKey The environment key to set or modify
+ * @param string $envValue The environment variable to set
+ * @return bool true on success or false on failure.
+ */
+function setEnvironmentValue(string $envKey, $envValue)
 {
-    $envFile = dirname(__FILE__, 3).'/.env';
-    $str = file_get_contents($envFile);
-
-    $str .= "\n"; // In case the searched variable is in the last line without \n
-    $keyPosition = strpos($str, "{$envKey}=");
-    $endOfLinePosition = strpos($str, PHP_EOL, $keyPosition);
-    $oldLine = substr($str, $keyPosition, $endOfLinePosition - $keyPosition);
-    $str = str_replace($oldLine, "{$envKey}={$envValue}", $str);
-    $str = substr($str, 0, -1);
-
-    $fp = fopen($envFile, 'w');
-    fwrite($fp, $str);
-    fclose($fp);
+    $str = "{$envKey}={$envValue}";
+    return putenv($str);
 }
 
-function getEnvironmentValue($envKey)
+/**
+ * Gets the variable from the env file
+ * @param string $envKey The environment variable to look for
+ * @return array|false|string Returns the value if found, otherwise returns false.
+ */
+function getEnvironmentValue(string $envKey): array|false|string
 {
-    $envFile = dirname(__FILE__, 3).'/.env';
-    $str = file_get_contents($envFile);
+    return getenv($envKey);
+}
+
 
-    $str .= "\n"; // In case the searched variable is in the last line without \n
-    $keyPosition = strpos($str, "{$envKey}=");
-    $endOfLinePosition = strpos($str, PHP_EOL, $keyPosition);
-    $oldLine = substr($str, $keyPosition, $endOfLinePosition - $keyPosition);
-    $value = substr($oldLine, strpos($oldLine, '=') + 1);
+/**
+ * Encrypt the given value
+ * @param mixed $value The variable to be encrypted
+ * @param bool $serialize If the encryption should be serialized
+ * @return string Returns the encrypted variable.
+ */
+function encryptSettingsValue(mixed $value, $serialize = true): string
+{
+    $appKey = getEnvironmentValue('APP_KEY');
+    $appKey = base64_decode(Str::after($appKey, 'base64:'));
+    $encrypter = new Encrypter($appKey, 'AES-256-CBC');
+    $encryptedKey = $encrypter->encrypt($value, $serialize);
+
+    return $encryptedKey;
+}
+
+/**
+ * Decrypt the given value
+ * @param mixed $payload The payload to be decrypted
+ * @param bool $unserialize If the encryption should be unserialized
+ * @return mixed Returns the decrypted variable on success, throws otherwise.
+ */
+
+function decryptSettingsValue(mixed $payload, $unserialize = true)
+{
+    $appKey = getEnvironmentValue('APP_KEY');
+    $appKey = base64_decode(Str::after($appKey, 'base64:'));
+    $encrypter = new Encrypter($appKey, 'AES-256-CBC');
+    $decryptedKey = $encrypter->decrypt($payload, $unserialize);
 
-    return $value;
+    return $decryptedKey;
 }
 
-function run_console($command)
+/**
+ * Run a shell command
+ * @param string $command The command string to run
+ * @return false|string|null Returns the result from the command.
+ */
+function run_console(string $command)
```

Comment:
This should be refactored. It does only support the bash shell which won't be present on windows or some specific linux distros. Rather use php shell_exec for example:
https://www.php.net/manual/en/function.shell-exec.php

Reply:
Are we really going to make sure people can host this dashboard on windows when even pterodactyl themselves do not offer support for windows hosting?

Reply:
no. we do not offer windows support. its never mentioned anywhere. we do somewhere say it works on ubuntu, debian and centos. everything else is not our business